DFS_2_Problem_1_numberIslands.java

```diff
@@ -0,0 +1,50 @@
+//Time Complexity : O(m*n) We might need to traverse through the whole matrix
+//Space Complexity : O(1)
```

Comment:
What about the recursive stack space? Worst case, the dfs goes O(MN) deep. So, space complexity can be O(MN) too